tests/testthat/test_leapfrog.R

```diff
@@ -0,0 +1,138 @@
+                                "pregprevlag", "entrantprev")
+  known_different_attributes <- c("rvec_ts")
+  attribuites_to_check <- setdiff(
+    attribute_names,
+    c(known_missing_attributes, known_different_attributes))
+
+  for (attr_name in attribuites_to_check) {
+    expected <- attr(mod_expected, attr_name)
+    received <- attr(mod, attr_name)
+    if (is.array("expected")) {
+      expect_equal(dim(received), dim(expected), info = attr_name)
+    } else {
+      expect_equal(length(received), length(expected), info = attr_name)
+    }
+  }
+}
+
+test_that("can run convert transmission input fp into leapfrog params", {
+  pjnz <- system.file("extdata", "testpjnz", "Botswana2018.PJNZ", package = "eppasm")
+  inputs <- prepare_spec_fit(pjnz, 2025.5)
+
+  fp <- prep_fp_fitmod(inputs$Urban, eppmod = "rhybrid")
+  fp <- stats::update(fp, list=fnCreateParam(theta_rhybrid, fp))
+
+  params <- fp_to_leapfrog_params(fp)
+
+  expect_equal(params$projection_start_year, 1970)
+  expect_equal(params$sim_years, 48)
+  expect_equal(params$incidence_model_choice, 1L)
+  expect_true(length(params$transmission_rate_hts) > 0)
+  expect_true(any(params$transmission_rate_hts > 0))
+  expect_equal(class(params), "specfp")
+})
+
+test_that("can convert output from leapfrog into mod data", {
+  pjnz <- system.file("extdata", "testpjnz", "Botswana2018.PJNZ", package="eppasm")
+  params <- leapfrog::prepare_leapfrog_parameters(pjnz, use_coarse_age_groups = TRUE)
+
+  n_years <- dim(params$Sx)[length(dim(params$Sx))]
+  output_years <- seq(params$projection_start_year, params$projection_start_year + n_years - 1)
+  leapfrog_result <- leapfrog::run_model(params, "HivCoarseAgeStratification", output_years = output_years)
+
+  params$sim_years <- length(output_years)
+  params$proj_years <- params$sim_years
+  mod <- leapfrog_result_to_mod(leapfrog_result, params)
+
+  fp <- prepare_directincid(pjnz)
+  mod_expected <- simmod(fp, VERSION = "C")
+
+  expect_mod_equivalent(mod, mod_expected)
+})
+
+test_that("can run simmod with leapfrog direct incidence data", {
+  pjnz <- system.file("extdata", "testpjnz", "Botswana2018.PJNZ", package = "eppasm")
+  fp <- prepare_directincid(pjnz)
+
+  mod <- simmod(fp, VERSION = "leapfrog")
+  mod_expected <- simmod(fp, version = "C")
+
+  expect_mod_equivalent(mod, mod_expected)
+})
+
+test_that("can run simmod with leapfrog transmission input data", {
+  pjnz <- system.file("extdata", "testpjnz", "Botswana2018.PJNZ", package = "eppasm")
+  inputs <- prepare_spec_fit(pjnz, 2018.5)
+
+  fp <- prep_fp_fitmod(inputs$Urban, eppmod = "rhybrid")
+  fp <- stats::update(fp, list=fnCreateParam(theta_rhybrid, fp))
+
+  mod <- simmod(fp, VERSION = "leapfrog")
+  mod_expected <- simmod(fp, VERSION = "C")
+
+  expect_mod_equivalent(mod, mod_expected)
+
+  # Assert outputs are not all 0
+  expect_true(any(attr(mod, "hivpop") > 0))
+  expect_true(any(attr(mod, "artpop") > 0))
+  expect_true(any(attr(mod, "infections") > 0))
```

Comment:
This test is failing at the moment but I am not sure why. I wonder if I am missing some input? I checked leapfrog and that is definitely setting values when running with transmission input. @jeffeaton anything obvious I am missing?